gui/web/src/components/atoms/Button/Button.js

```diff
@@ -55,17 +52,27 @@ class Button extends Component {
       return
     }
     
+    if (this.props.eventName === "" || this.props.eventName === undefined) {
+      return
+    }
+    
     var _this = this
-    // TODO DS populate the eventData.
-    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, {}).then(resp => {
+    var eventData = {
+      'eventName': this.props.eventName,
+      'type': 'generic',
+      'component': 'button'
+    }
+    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, eventData).then(resp => {
       if (!_this._asyncRequests["sendMetricEvent"]) {
         // if it has been deleted it means we don't want to process the result
         return
       }
       
       delete _this._asyncRequests["sendMetricEvent"];
       
-      // TODO DS Determine how to process resp
+      if (resp["success"] === undefined) {
+        console.log(resp["error"])
+      }
```

Comment:
add a comment like this after this so we know we have explicitly chosen to ignore the else case:
`// else do nothing on success`